harness/determined/common/api/request.py

```diff
@@ -128,8 +67,16 @@ def do_request(
     body: Optional[Dict[str, Any]] = None,
     headers: Optional[Dict[str, str]] = None,
     authenticated: bool = True,
+    auth: Optional[authentication.Authentication] = None,
+    cert: Optional[certs.Cert] = None,
     stream: bool = False,
 ) -> requests.Response:
+    # If no explicit Authentication object was provided, use the cli's singleton Authentication.
+    if auth is None:
+        auth = authentication.cli_auth
```

Comment:
This `cli_auth` is actually used by non-cli functions?

Reply:
Yes, but that should change in the near future.  As far as I know, only the data layer and the profiler currently make API calls through here.